src/vivarium_gates_nutrition_optimization/components/pregnancy.py

```diff
@@ -16,9 +18,88 @@ def __init__(self, cause, *args, **kwargs):
 
 
 class PregnantState(DiseaseState):
+    @property
+    def columns_created(self):
+        return [
+            self.event_time_column,
+            self.event_count_column,
+            "pregnancy_term_outcome",
+            "pregnancy_duration",
+            # "pregnancy_birth_outcome",
+        ]
+
     def setup(self, builder: Builder):
         super().setup(builder)
         self.time_step = builder.time.step_size()
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.partial_term_probs = builder.value.register_value_producer(
+            "partial_term_probabilities",
+            source=builder.lookup.build_table(
+                get_partial_term_probability_data(builder),
+                key_columns=["sex"],
+                parameter_columns=["age", "year"],
+            ),
+        )
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        for transition in self.transition_set:
+            if transition.start_active:
+                transition.set_active(pop_data.index)
+
+        pop_events = self.get_initial_event_times(pop_data)
+        pregnancy_term_outcomes_and_durations = self.sample_pregnancy_terms_and_durations(
+            pop_data
+        )
+        pop_update = pd.concat([pop_events, pregnancy_term_outcomes_and_durations], axis=1)
+        self.population_view.update(pop_update)
+
```

Comment:
I assume, this is working, but I would have expected you need to make a `builder.population.initializes_simulants` call like [here](https://github.com/ihmeuw/vivarium_gates_iv_iron/blob/862b057457cea2410e8d24c2c9c38c43419cd8b2/src/vivarium_gates_iv_iron/components/pregnancy.py#L146-L151). It doesn't look like it happens in the superclass's setup either.

Reply:
this happens in the setup for the supersuperclass `BaseDiseaseState`